source/interprocedural_analyses/taint/modelParser.ml

```diff
@@ -3220,52 +3299,96 @@ let rec parse_statement
-                unexpected_models_clause;
-              } ->
-      parse_find_clause ~path find_clause
-      |> as_result_error_list
-      >>= fun find ->
-      parse_where_clause ~path ~find_clause:find where_clause
-      |> as_result_error_list
-      >>= fun where ->
-      parse_model_clause
-        ~path
-        ~taint_configuration
-        ~find_clause:find
-        ~is_object_target:(not (ModelQuery.Find.is_callable find))
-        model_clause
-      >>= check_write_to_cache_models ~path ~location ~where
-      |> as_result_error_list
-      >>= fun models ->
-      parse_output_models_clause ~name ~path expected_models_clause
-      >>= fun expected_models ->
-      parse_output_models_clause ~name ~path unexpected_models_clause
-      >>| fun unexpected_models ->
-      [
-        ParsedQuery
-          {
-            ModelQuery.find;
-            where;
-            models;
-            name;
-            logging_group_name;
-            path;
-            location;
-            expected_models;
-            unexpected_models;
-          };
-      ]
+      let result =
+        ModelQueryArguments.parse_arguments ~path ~location arguments
+        |> as_result_error_list
+        >>= fun {
+                  ModelQueryArguments.name;
+                  logging_group_name;
+                  find_clause;
+                  where_clause;
+                  model_clause;
+                  expected_models_clause;
+                  unexpected_models_clause;
+                } ->
+        parse_find_clause ~path find_clause
+        |> as_result_error_list
+        >>= fun find ->
+        parse_where_clause ~path ~find_clause:find where_clause
+        |> as_result_error_list
+        >>= fun where ->
+        parse_model_clause
+          ~path
+          ~taint_configuration
+          ~find_clause:find
+          ~is_object_target:(not (ModelQuery.Find.is_callable find))
+          model_clause
+        >>= check_write_to_cache_models ~path ~location ~where
+        |> as_result_error_list
+        >>= fun models ->
+        parse_output_models_clause ~name ~path expected_models_clause
+        >>= fun expected_models ->
+        parse_output_models_clause ~name ~path unexpected_models_clause
+        >>| fun unexpected_models ->
+        [
+          ParsedQuery
+            {
+              ModelQuery.find;
+              where;
+              models;
+              name;
+              logging_group_name;
+              path;
+              location;
+              expected_models;
+              unexpected_models;
+            };
+        ]
+      in
+      [result]
+  | {
+   Node.value =
+     If
+       {
+         If.body;
+         If.test =
+           {
+             Node.value =
+               ComparisonOperator
+                 {
+                   ComparisonOperator.right = { Node.value = Tuple tuple; _ };
+                   ComparisonOperator.operator;
+                   _;
```

Comment:
This means we silently assume the left hand side is `sys.version`? Not a fan of this.
That would mean we accept things like `if "hello" < (3, 6, 0)`
Could we only accept `sys.version`? It should be an `Identifier` AST node.

Reply:
Okay, we can do that!

Reply:
@arthaud also I checked the return value of the real sys.version, which is type string.. Just a heads up that we are deviating a bit from the expected behavior and could use something like `python_version`